### PR TITLE
refactor: v2 recurring bookings allow specifying recurrence count

### PR DESCRIPTION
**Pull request by supalarry**
Linear [CAL-4437](https://linear.app/calcom/issue/CAL-4437/platform-refactor-bookings-recurring)

### CODE REVIEW

File: apps/api/v2/src/ee/bookings/2024-08-13/services/input.service.ts [hunk 1]
```diff
@@ -179,8 +179,10 @@ export class InputBookingsService_2024_08_13 {
     // note(Lauris): timeBetween 0=yearly, 1=monthly and 2=weekly
     const timeBetween = occurrence.freq;
 
-    if (!repeatsTimes) {
-      throw new Error("Repeats times is required");
+    if (inputBooking.recurrenceCount && inputBooking?.recurrenceCount > occurrence.count) {
```
Comment (ThyMinimalDev) [inline apps/api/v2/src/ee/bookings/2024-08-13/services/input.service.ts:182]: NIT: no need for ? operator

**Review by ThyMinimalDev — DISMISSED** [review]
looking good, one tiny nit

File: apps/api/v2/src/ee/bookings/2024-08-13/services/input.service.ts (hunk 1 above)
Comment (supalarry) [reply apps/api/v2/src/ee/bookings/2024-08-13/services/input.service.ts]: Well spotted - removed in [Update input.service.ts](https://github.com/calcom/cal.com/pull/16989/commits/8bd4d0fec067913c61023b40cd3c56baa4631e45)

**Review by ibex088 — DISMISSED** [review]

**Review by ThyMinimalDev — APPROVED** [review]
looking good

**Pull request merged**